app/javascript/packages/document-capture/components/file-input.tsx

```diff
@@ -161,11 +218,13 @@ function FileInput(props, ref) {
   /**
    * In response to a file input change event, confirms that the file is valid before calling
    * `onChange`.
-   *
-   * @param {import('react').ChangeEvent<HTMLInputElement>} event Change event.
    */
-  function onChangeIfValid(event) {
-    const file = /** @type {FileList} */ (event.target.files)[0];
+  function onChangeIfValid(event: ReactChangeEvent<HTMLInputElement>) {
+    if (!event.target.files) {
+      return;
+    }
+
+    const file: File = event.target.files[0];
```

Comment:
May have been able to keep this as a single line, though I think splitting it out as you've done could help with the clarity.

```suggestion
    const file = event.target.files?.[0];
```